com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -546,6 +574,22 @@ def cancel(self, force=False):
             return True
         raise NotImplementedError('Job.cancel()')
 
+    def store_logs(self, filename=None):
```

Comment:
How does this relate to `get_application_logs` ?

Reply:
The store_logs() method uses the same REST API as get_application_logs(). The difference in the signature is that it has only one optional argument, which allows the caller to specify the tarball's full path. The difference in the implementation is that all of the download code is shared with get_console_log() and get_application_trace(). We certainly don't need both store_logs(), but I really don't think the additional characters get_application_logs() inserts into the filename are a good idea.

Reply:
We need a single mechanism in the api, with consistent naming, we don't just want to add alternatives to support a different style.

Reply:
Can you separate out this multiple changes into isolated prs. Submit job is separate functionality to any change in the `get_application_logs` api.

Reply:
Sure. I've replaced this pull request with two more, one for submit_job() and one for store_logs()